R/examine-constraints.R

```diff
@@ -119,16 +125,19 @@ check_pk_constraints <- function(dm) {
     left_join(tbl_is_pk, by = c("table", "column"))
 }
 
-check_fk_constraints <- function(dm) {
+check_fk_constraints <- function(dm, progress = NA) {
   fks <- dm_get_all_fks_impl(dm)
   pts <- pull(fks, parent_table) %>% map(tbl_impl, dm = dm)
   cts <- pull(fks, child_table) %>% map(tbl_impl, dm = dm)
   fks_tibble <-
     mutate(fks, t1 = cts, t2 = pts) %>%
     select(t1, t1_name = child_table, colname = child_fk_cols, t2, t2_name = parent_table, pk = parent_key_cols)
+
+  ticker <- new_ticker("checking pk constraints", nrow(fks_tibble), progress = progress)
```

Comment:
This needs to be "fk constraints"? Should we spell out "primary key" and "foreign key"?